tests/common.py

```diff
@@ -379,11 +379,36 @@ def async_fire_mqtt_message(hass, topic, payload, qos=0, retain=False):
 fire_mqtt_message = threadsafe_callback_factory(async_fire_mqtt_message)
 
 
+@ha.callback
+def async_fire_time_changed_exact(
+    hass: HomeAssistant, datetime_: datetime | None = None, fire_all: bool = False
+) -> None:
+    """Fire a time changed event at an exact microsecond.
+
+    Consider that its not possible to actually achieve an exact microsecond
+    in production as the event loop is not precise enough. If your code
+    relies on this level of precision, consider a different approach
+    as this is only for testing.
+    """
+    if datetime_ is None:
+        utc_datetime = date_util.utcnow()
+    else:
+        utc_datetime = date_util.as_utc(datetime_)
+
+    _async_fire_time_changed(hass, utc_datetime, fire_all)
+
+
 @ha.callback
 def async_fire_time_changed(
-    hass: HomeAssistant, datetime_: datetime = None, fire_all: bool = False
+    hass: HomeAssistant, datetime_: datetime | None = None, fire_all: bool = False
 ) -> None:
-    """Fire a time changed event."""
+    """Fire a time changed event.
+
+    This function will ensure microseconds at at least 500000
+    to account for the synchronization repeating listeners.
```

Comment:
This line doesn't sound right. Is a word missing?

Reply:
https://github.com/home-assistant/core/pull/82349